lib/index.js

```
@@ -228,15 +230,29 @@ module.exports = class DockerComposeParser {
 
   /**
    * Get main container for docker compose cluster. Picks the first container
-   * with a `build` property.
+   * with a `build` property that isn't from github, unless it only has github, then it picks that first one
    *
    * @param {Object<Key:Service>} services - Object with service names as keys and service objects as values
-   * @returns {String}
+   * @returns {String||Undefined} Returns the key name of the main instance, or undefined if it's not there
    */
   static _getMainName (services) {
-    return Object.keys(services)
-      .map(key => services[key].build && key)
-      .find(key => !!key)
+    const buildLists = Object.keys(services)
+      .filter(key => !!services[key].build)
+      .reduce((buildLists, key) => {
+        const buildObj = services[key].build
+        if (typeof (buildObj) === 'string' && GITHUB_REGEX.test(buildObj)) {
+          buildLists.links.push(key)
+        } else {
+          buildLists.builds.push(key)
+        }
+        return buildLists
+      }, { builds: [], links: [] })
+
+    if (buildLists.builds.length) {
+      return buildLists.builds[0]
+    } else if (buildLists.links.length) {
```

Feel like we should throw an error if both of these fail